editor/src/components/canvas/canvas-strategies/strategies/shared-absolute-resize-strategy-helpers.ts

```diff
@@ -274,3 +297,141 @@ function pinsFromEdgePosition(edgePosition: EdgePosition): Array<LayoutEdgeProp>
+
+    const elementMetadata = MetadataUtils.findElementByElementPath(
+      startingMetadata,
+      selectedElement,
+    )
+    const elementParentBounds =
+      elementMetadata?.specialSizeMeasurements.immediateParentBounds ?? null
+    const elementParentFlexDirection =
+      elementMetadata?.specialSizeMeasurements.parentFlexDirection ?? null
+    const elementGlobalFrame = nullIfInfinity(elementMetadata?.globalFrame ?? null)
+
+    if (element != null && isJSXElement(element)) {
+      const elementResult = createResizeCommands(
+        element,
+        selectedElement,
+        edgePosition,
+        movement,
+        elementGlobalFrame,
+        elementParentBounds,
+        elementParentFlexDirection,
+      )
+      commands.push(...elementResult.commands)
+      if (elementResult.intendedBounds != null) {
+        intendedBounds = addOrMergeIntendedBounds(
+          intendedBounds,
+          originalFrame,
+          elementResult.intendedBounds,
+        )
+      }
+    }
+  })
+
+  return {
+    commands: commands,
+    intendedBounds: intendedBounds,
+  }
+}
+
+export function resizeInspectorStrategy(
+  projectContents: ProjectContentTreeRoot,
+  originalFrame: CanvasRectangle,
+  edgePosition: EdgePosition,
+  movement: CanvasVector,
+): InspectorStrategy {
+  return {
+    name: 'Resize by pixels',
+    strategy: (
+      metadata: ElementInstanceMetadataMap,
+      selectedElements: Array<ElementPath>,
+      _elementPathTree: ElementPathTrees,
+      _allElementProps: AllElementProps,
+    ): Array<CanvasCommand> | null => {
+      let commands: Array<CanvasCommand> = []
+      const changeBoundsResult = changeBounds(
+        projectContents,
+        metadata,
+        selectedElements,
+        originalFrame,
+        [],
+        edgePosition,
+        movement,
+      )
+      commands.push(...changeBoundsResult.commands)
+      commands.push(
+        pushIntendedBoundsAndUpdateGroups(changeBoundsResult.intendedBounds, 'starting-metadata'),
+      )
+      commands.push(setElementsToRerenderCommand(selectedElements))
+      return commands
+    },
+  }
+}
+
+export function directResizeInspectorStrategy(
+  projectContents: ProjectContentTreeRoot,
+  widthOrHeight: 'width' | 'height',
+  newPixelValue: number,
+): InspectorStrategy {
+  return {
+    name: 'Resize to pixel size',
+    strategy: (
+      metadata: ElementInstanceMetadataMap,
+      selectedElements: Array<ElementPath>,
+      _elementPathTree: ElementPathTrees,
+      _allElementProps: AllElementProps,
+    ): Array<CanvasCommand> | null => {
+      let commands: Array<CanvasCommand> = []
+      for (const selectedElement of selectedElements) {
+        const edgePosition: EdgePosition =
+          widthOrHeight === 'width' ? EdgePositionRight : EdgePositionBottom
+        const elementMetadata = MetadataUtils.findElementByElementPath(metadata, selectedElement)
+        const originalFrame = elementMetadata?.globalFrame
```

Comment:
should this be the localFrame instead? are we using localFrame anywhere anymore?

Reply:
I went back and forth a little on this, IIRC the pre-existing code pushed me more in the direction of the global canvas frame.